src/backend/distributed/executor/multi_utility.c

```diff
@@ -1474,8 +1422,8 @@ ErrorIfUnstableCreateOrAlterExtensionStmt(Node *parsetree)
 		 * from the citus.control file. In case a new default is available, we
 		 * will force a compatibility check of the latest available version.
 		 */
-		availableExtensionVersion = NULL;
-		ErrorIfAvailableVersionMismatch();
+		citusVersionKnownCompatible = false;
+		CheckCitusVersion(ERROR);
```

Comment:
Yes, basically we do not want users to run ALTER EXTENSION UPDATE before restarting the database first.